File: VB/XtraReport1.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace ExportGroups
    Partial Public Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Public groups As New ArrayList()
        Private startPageIndex As Integer = 0
        Private Sub xrLabel7_PrintOnPage(ByVal sender As Object, ByVal e As PrintOnPageEventArgs) Handles xrLabel7.PrintOnPage
            startPageIndex = e.PageIndex
        End Sub

        Private Sub xrLabel8_PrintOnPage(ByVal sender As Object, ByVal e As PrintOnPageEventArgs) Handles xrLabel8.PrintOnPage
            groups.Add(New Groups(startPageIndex, e.PageIndex))
        End Sub
    End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace ExportGroups {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        public ArrayList groups = new ArrayList();
        int startPageIndex = 0;
        private void xrLabel7_PrintOnPage(object sender, PrintOnPageEventArgs e) {
            startPageIndex = e.PageIndex;
        }

        private void xrLabel8_PrintOnPage(object sender, PrintOnPageEventArgs e) {
            groups.Add(new Groups(startPageIndex, e.PageIndex));
        }
    }
}
